velox/dwio/parquet/reader/CompressionQpl.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <sys/time.h>
+
+#include "CompressionQpl.h"
+#include "velox/dwio/common/BufferUtil.h"
+
+#define MAX_COMPRESS_CHUNK_SIZE 262144
+#define MAX_DECOMPRESS_CHUNK_SIZE 262144
+#define QPL_MAX_TRANS_SIZE 2097152
+#define JOB_BUFFER_SIZE 820 * 1024
+
+thread_local bool initialize_job = false;
+thread_local uint8_t job_buffer[JOB_BUFFER_SIZE];
+thread_local qpl_job* job;
```

Comment:
You can put all these inside `Qplcodec` and have `thread_local Qplcodec` here, then access it using a static method `static Qplcodec& Qplcodec::getInstance()`.  In `~Qplcodec()` you can release the resource.